datafusion/core/Cargo.toml

```diff
@@ -60,9 +62,10 @@ async-trait = "0.1.41"
 avro-rs = { version = "0.13", features = ["snappy"], optional = true }
 chrono = { version = "0.4", default-features = false }
 datafusion-common = { path = "../common", version = "7.0.0", features = ["parquet"] }
-datafusion-data-access = { path = "../../data-access", version = "1.0.0" }
+datafusion-data-access = { git = "https://github.com/apache/arrow-datafusion.git", rev = "41b4e491663029f653e491b110d0b5e74d08a0b6" }
```

Comment:
This change seems a little suspicious to me (as the data access crate should be picked up locally)

Reply:
Hi @alamb, with the datafusion-data-access module spitted out, now the dependencies are as follows:
<pre>
                      ballista
                           ⬇️ 
                    datafusion   ➡️   objectstore-hdfs
                           ⬇️             ⬇️ 
                       datafusion-data-access
</pre>

Therefore, there's no cyclic dependency. Maybe it's better to split the datafusion-data-access into a separate repository and make a first version of release. Then both the datafusion and objectstore-hdfs can depend on the specified version of datafusion-data-access rather than leveraging the rev way.

Reply:
There is no circular dependency in the cargo dependency graph, but there is a circular dependency on how these crates gets released. The data access crate needs to be released together with datafusion right now, which is what's causing the problem.

Reply:
Yes. Since the data access crate is relatively stable, I'm not sure whether it's necessary to release it together with datafusion. How about splitting it to another repository, for example to the https://github.com/datafusion-contrib?

Reply:
yeah, i am leaning towards moving the data access crate into contrib as well

Reply:
Isn't `datafusion-data-access` a core part of datfusion though. The `ObjectStore` trait is baked in everywhere. It seems a bit weird to move that into a contrib module. 

Would it be possible to leverage something like https://github.com/apache/arrow-datafusion/pull/1881 to load contrib `ObjectStores` as plugins?